Hardware/Hardware/CardReader/Charge.swift

```diff
@@ -0,0 +1,27 @@
+/// A struct representing a charge.
+public struct Charge {
+    /// The unique identifier for the charge.
+    let identifier: String
+
+    /// The amount of the charge.
+    let amount: Int
+
+    /// The currency of the charge
+    let currency: String
+
+    /// The status of a charge
+    let status: ChargeStatus
+
+    /// A string describing the charge.
+    let description: String
+
+    /// Metadata associated with the charge.
+    let metadata: [AnyHashable: Any]?
+}
+
+
+extension Charge: Equatable {
+    public static func ==(lhs: Charge, rhs: Charge) -> Bool {
+        return lhs.identifier == rhs.identifier
+    }
```

Comment:
Same here, for value-types, I think this should be something like `Identifiable` and not `Equatable`. WDYT?

Reply:
Oh, like use Identifiable as a way to decouple identity from state? Yep, that makes sense, I'll update, thanks!